internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,277 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			return ctrl.Result{Requeue: true}, nil
 		}
 	}
+
 	logger.V(1).Info("NonAdminBackup Reconcile exit")
 	return ctrl.Result{}, nil
 }
 
-// init initializes the Status.Phase from the NonAdminBackup.
+// setStatusAndConditionForDeletionAndCallDelete updates the NonAdminBackup status and conditions
+// to reflect that deletion has been initiated, and triggers the actual deletion if needed.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime.
+//   - logger: Logger instance for logging messages.
+//   - nab: Pointer to the NonAdminBackup object being processed.
+//
+// Returns:
+//   - bool: true if reconciliation should be requeued, false otherwise
+//   - error: any error encountered during the process
+func (r *NonAdminBackupReconciler) setStatusAndConditionForDeletionAndCallDelete(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseDeleting)
+	updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
+		metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminConditionDeleting),
+			Status:  metav1.ConditionTrue,
+			Reason:  "DeletionPending",
+			Message: "backup accepted for deletion",
+		},
+	)
+	if updatedPhase || updatedCondition {
+		if err := r.Status().Update(ctx, nab); err != nil {
+			logger.Error(err, statusUpdateError)
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackup status marked for deletion")
+		return true, nil
+	}
+	if nab.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.V(1).Info("Marking NonAdminBackup for deletion", nameString, nab.Name)
+		if err := r.Delete(ctx, nab); err != nil {
+			logger.Error(err, "Failed to call Delete on the NonAdminBackup object")
+			return false, err
+		}
+		return true, nil // Requeue to allow deletion to proceed
+	}
+	return false, nil
+}
+
+// setStatusForDirectKubernetesAPIDeletion updates the status and conditions when a NonAdminBackup
+// is deleted directly through the Kubernetes API. Only updates status and conditions
+// if the NAB finalizer exists.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime
+//   - logger: Logger instance
+//   - nab: NonAdminBackup being deleted
+//
+// Returns:
+//   - bool: true if status was updated
+//   - error: any error encountered
+func (r *NonAdminBackupReconciler) setStatusForDirectKubernetesAPIDeletion(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseDeleting)
+	updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
+		metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminConditionDeleting),
+			Status:  metav1.ConditionTrue,
+			Reason:  "DeletionPending",
+			Message: "backup deletion requires VeleroBackup deletion or finalizer removal",
```

Comment:
here message is misleading, not?

If I do `oc delete` and deletes Velero backup, NAB will not be deleted

what about something like: 
> backup deletion requires setting spec.deleteBackup or spec.forceDeleteBackup to true or finalizer removal

Reply:
Fixed.